### PR TITLE
Add MicroProfile LRA subsystem tests

### PR DESCRIPTION
**Pull request by xstefank**
Follow up for https://issues.redhat.com/browse/WFLY-14869

Please make sure your PR meets the following requirements:
- [x] Pull Request contains a description of the changes
- [x] Pull Request does not include fixes for multiple issues/topics
- [x] Code is formatted, imports ordered, code compiles and tests are passing
- [ ] Link to the passing job is provided
- [x] Code is self-descriptive and/or documented
- [x] Description of the tests scenarios is included (see #46)

### CODE REVIEW

**Comment by xstefank** [comment]
cc/ @fabiobrz

File: microprofile-lra/src/test/java/org/jboss/eap/qe/microprofile/lra/LRAPropagationTest.java [hunk 1]
```diff
@@ -0,0 +1,373 @@
...
+public class LRAPropagationTest {
+
+    @ArquillianResource
+    private URL baseApplicationUrl;
+
+    @Deployment(testable = false)
+    public static Archive<?> deployment() {
+        return ShrinkWrap.create(WebArchive.class, LRAPropagationTest.class.getSimpleName() + ".war")
+                .addPackages(true, LRAController.class.getPackage())
+                .addClasses(EnableLRASetupTask.class, MicroProfileServerSetupTask.class, ServerSetupTask.class)
```
[the remark this answers is not in the document]
Comment (xstefank) [reply microprofile-lra/src/test/java/org/jboss/eap/qe/microprofile/lra/LRAPropagationTest.java:50]: They get rid of this warning that is printed when the test runs:

```
07:47:51,529 WARN  [org.jboss.modules.define] (Weld Thread Pool -- 10) Failed to define class org.jboss.eap.qe.microprofile.lra.EnableLRASetupTask in Module "deployment.LRAPropagationTest.war" from Service Module Loader: java.lang.NoClassDefFoundError: Failed to link org/jboss/eap/qe/microprofile/lra/EnableLRASetupTask (Module "deployment.LRAPropagationTest.war" from Service Module Loader): org/jboss/eap/qe/microprofile/tooling/server/configuration/arquillian/MicroProfileServerSetupTask
	at java.base/java.lang.ClassLoader.defineClass1(Native Method)
	at java.base/java.lang.ClassLoader.defineClass(ClassLoader.java:1017)
	at java.base/java.lang.ClassLoader.defineClass(ClassLoader.java:1096)
	at org.jboss.modules.ModuleClassLoader.doDefineOrLoadClass(ModuleClassLoader.java:351)
	at org.jboss.modules.ModuleClassLoader.defineClass(ModuleClassLoader.java:482)
	at org.jboss.modules.ModuleClassLoader.loadClassLocal(ModuleClassLoader.java:276)
	at org.jboss.modules.ModuleClassLoader$1.loadClassLocal(ModuleClassLoader.java:79)
	at org.jboss.modules.Module.loadModuleClass(Module.java:765)
	at org.jboss.modules.ModuleClassLoader.findClass(ModuleClassLoader.java:192)
	at org.jboss.modules.ConcurrentClassLoader.performLoadClassUnchecked(ConcurrentClassLoader.java:410)
	at org.jboss.modules.ConcurrentClassLoader.performLoadClass(ConcurrentClassLoader.java:398)
	at org.jboss.modules.ConcurrentClassLoader.loadClass(ConcurrentClassLoader.java:116)
	at org.jboss.as.weld@30.0.1.Final//org.jboss.as.weld.WeldModuleResourceLoader.classForName(WeldModuleResourceLoader.java:51)
	at org.jboss.weld.core@5.1.2.Final//org.jboss.weld.bootstrap.AnnotatedTypeLoader.loadClass(AnnotatedTypeLoader.java:86)
	at org.jboss.weld.core@5.1.2.Final//org.jboss.weld.bootstrap.AnnotatedTypeLoader.loadAnnotatedType(AnnotatedTypeLoader.java:65)
	at org.jboss.weld.core@5.1.2.Final//org.jboss.weld.bootstrap.FastAnnotatedTypeLoader.loadAnnotatedType(FastAnnotatedTypeLoader.java:112)
	at org.jboss.weld.core@5.1.2.Final//org.jboss.weld.bootstrap.BeanDeployer.addClass(BeanDeployer.java:87)
	at org.jboss.weld.core@5.1.2.Final//org.jboss.weld.bootstrap.ConcurrentBeanDeployer$1.doWork(ConcurrentBeanDeployer.java:55)
	at org.jboss.weld.core@5.1.2.Final//org.jboss.weld.bootstrap.ConcurrentBeanDeployer$1.doWork(ConcurrentBeanDeployer.java:52)
	at org.jboss.weld.core@5.1.2.Final//org.jboss.weld.executor.IterativeWorkerTaskFactory$1.call(IterativeWorkerTaskFactory.java:62)
	at org.jboss.weld.core@5.1.2.Final//org.jboss.weld.executor.IterativeWorkerTaskFactory$1.call(IterativeWorkerTaskFactory.java:55)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:829)
	at org.jboss.threads@2.4.0.Final//org.jboss.threads.JBossThread.run(JBossThread.java:513)
```

So I prefer to keep them.

[the remark this answers is not in the document]
Comment (fabiobrz) [reply microprofile-lra/src/test/java/org/jboss/eap/qe/microprofile/lra/LRAPropagationTest.java:50]: +1, thanks

**Review by fabiobrz — APPROVED** [review]
LGTM, thanks @xstefank - Merging.

**Pull request merged**